2005-06-13  Sergey Poznyakoff  <gray@Mirddin.farlep.net>

	* src/create.c (dump_file0): Check for is_avoided_name()
	first. Fixes bug reported by Martin Lohmeier
	<martin@mein-horde.de>

--- src/create.c	21 May 2005 23:10:42 -0000	1.99
+++ src/create.c	13 Jun 2005 19:09:45 -0000	1.100
@@ -1414,6 +1414,8 @@ dump_file0 (struct tar_stat_info *st, ch
       return;
     }
 
+  if (is_avoided_name (p))
+    return;
   if (S_ISDIR (st->stat.st_mode))
     {
       dump_dir (st, top_level, parent_device);
@@ -1421,8 +1423,6 @@ dump_file0 (struct tar_stat_info *st, ch
 	utime (p, &restore_times);
       return;
     }
-  else if (is_avoided_name (p))
-    return;
   else
     {
       /* Check for multiple links.  */
